range_queries/heavy_light_decomposition.cpp

```diff
@@ -237,7 +237,7 @@ template <typename X> class Tree {
     if (a == b) {
       return a;
     }
-    for (int k = 0; k < t_maxlift; k++) {
+    for (int k = t_maxlift - 1; k >= 0; k--) {
```

Comment:
any particular reason for the change or loop direction?

Reply:
The forward loop is wrong for finding the lca. I noticed it now when I was going through the code once again. Taking the highest set bit first will guarantee ending up on the lca, but taking the least set bit first might not.

I can give an example if required.

Reply:
👍 Please add that case in the self-test. Thank you

Reply:
I have added the test case as `test_case3()`, with more generic queries.